lib/sprockets/resolve.rb

```diff
@@ -153,6 +153,12 @@ def parse_accept_options(mime_type, types)
       end
 
       def path_matches(load_path, logical_name, logical_basename)
+        @@gorilla_djungle_banana_cache ||= {}
+        app_root = Sprockets.check_modified_root
+        is_in_app = load_path[0...app_root.size] == app_root
```

Comment:
Might make sense to stringify `app_root` here. That config option can be `nil`.

Reply:
Actually, that method call is used in several places. Might make sense to always return a string from it instead.

Reply:
```
is_in_app = load_path.start_with?(app_root.to_s)
```